datalad_xnat/platform.py

```diff
@@ -82,15 +87,23 @@ def __init__(self, url, credential):
                         password=auth['password'] or None)
             session.auth = (auth['user'], auth['password'])
 
-        # now check of auth works (if any is needed)
+        # now check if auth works (if any is needed)
         # TODO check that we have anonymous OR user/pass
         try:
             session.post(self._get_api('session_token')).raise_for_status()
+        except HTTPError as e:
+            # TODO: Accessing non-public requests.status_codes._codes isn't nice.
+            #       But the respective LookupDict seems only useful to look up a
+            #       code based on a name, not the other way around. That is,
+            #       however, what we need here to be more informative than
+            #       XNAT's "Client Error for url XXX" for an 401 for example.
+            msg = e.response.reason or \
+                  status_codes._codes[e.response.status_code][0]
```

Comment:
I think it would be better to use https://docs.python.org/3/library/http.html#http.HTTPStatus

Reply:
`{i.value: i.phrase for i in http.HTTPStatus}`

Reply:
Done. Also looked in `.description` as an alternative, but that phrasing requires "protocol awareness" - not more useful for users.